client/api/openapi.yaml

```diff
@@ -5670,6 +5670,190 @@ components:
+          type: array
+        secretId:
+          example: 62bc3c75-91fb-4670-bad4-24b401a9deac
+          type: string
+        secretName:
+          example: my-aws-secret
+          type: string
+        sshSecretId:
+          example: 62bc3c75-91fb-4670-bad4-24b401a9deac
+          type: string
+        scaleOptions:
+          $ref: '#/components/schemas/ScaleOptions'
+        type:
+          type: string
+      required:
+      - name
+      - type
+      type: object
+    Feature:
+      example:
+        kind: logging
+        params:
+          enabled: true
+      properties:
+        kind:
+          type: string
+        params:
+          additionalProperties: true
+          type: object
+      type: object
+    ScaleOptions:
+      properties:
+        enabled:
+          type: boolean
+        desiredCpu:
+          format: double
+          minimum: 1
+          type: number
+        desiredMem:
+          format: double
+          minimum: 1
+          type: number
+        desiredGpu:
+          format: int32
+          minimum: 0
+          type: integer
+        onDemandPct:
+          format: int32
+          maximum: 100
+          minimum: 0
+          type: integer
+        excludes:
+          items:
+            type: string
+          type: array
+        keepDesiredCapacity:
+          type: boolean
+      type: object
+    CreatePKEClusterRequestBase:
+      allOf:
+      - $ref: '#/components/schemas/CreateClusterRequestBase'
+      - properties:
+          kubernetes:
+            properties:
+              version:
+                example: v1.12.2
+                type: string
+              rbac:
+                type: boolean
+              cri:
+                properties:
+                  runtime:
+                    type: string
+                  runtimeConfig:
+                    additionalProperties: true
+                    type: object
+                type: object
+              network:
+                properties:
+                  serviceCIDR:
+                    example: 10.32.0.0/24
+                    type: string
+                  podCIDR:
+                    example: 10.200.0.0/16
+                    type: string
+                  provider:
+                    enum:
+                    - weave
+                    type: string
+                  providerConfig:
+                    additionalProperties: true
+                    type: object
+                type: object
+            type: object
+        type: object
+    CreatePKEOnAzureClusterRequest:
+      allOf:
+      - $ref: '#/components/schemas/CreatePKEClusterRequestBase'
+      - properties:
+          location:
+            example: westeurope
+            type: string
+          resourceGroup:
+            example: my-resource-group-1
+            type: string
+          network:
+            properties:
+              name:
+                type: string
+              cidr:
```

Comment:
shouldn't this be a collection of cidrs?

Reply:
I'm not sure if we'd like to support creating and automatically allocating subranges from a multi-range network. When the network already exists only its name has to be specified as reference.